Hunting Queries/SecurityEvent/RareProcessPath.yaml

```diff
@@ -0,0 +1,66 @@
+  | where NewProcessName !endswith ":\\WindowsAzure\\SecAgent\\WaSecAgentProv.exe" 
+  //| where ParentProcessName !has ":\\WindowsAzure\\GuestAgent_" and NewProcessName !has ":\\WindowsAzure\\GuestAgent_"
+  | where ParentProcessName !has ":\\WindowsAzure\\WindowsAzureNetAgent_" and NewProcessName !has ":\\WindowsAzure\\WindowsAzureNetAgent_"
+  | where ParentProcessName !has ":\\ProgramData\\Microsoft\\Windows Defender\\platform\\" and ParentProcessName !endswith "\\MpCmdRun.exe" 
+  | where NewProcessName !has ":\\ProgramData\\Microsoft\\Windows Defender\\platform\\" and NewProcessName !endswith "\\MpCmdRun.exe" 
+  | where NewProcessName !has ':\\Program Files\\Microsoft Monitoring Agent\\Agent\\'
+  // filter out common randomly named paths and files
+  | where not(NewProcessName matches regex @"\\TRA[0-9A-Fa-f]{3}\.tmp")
+  | where not(NewProcessName matches regex @"\\TRA[0-9A-Fa-f]{4}\.tmp")
+  | where not(NewProcessName matches regex @"Installer\\MSI[0-9A-Fa-f]{3}\.tmp")
+  | where not(NewProcessName matches regex @"Installer\\MSI[0-9A-Fa-f]{4}\.tmp")
+  | where not(NewProcessName matches regex @"\\Windows\\Temp\\[0-9A-Za-z-]*\\DismHost\.exe")
+  | where not(NewProcessName matches regex @"\\Users\\[0-9A-Za-z-_~\.]*\\AppData\\Local\\Temp\\[0-9A-Za-z-]*\\DismHost\.exe")
+  | where not(NewProcessName matches regex @"\\Windows\\Temp\\[0-9A-Za-z-]*\\MpSigStub\.exe")
+  | where not(NewProcessName matches regex @"\\[0-9A-Za-z]*\\amd64\\setup\.exe") and (ParentProcessName !has ":\\Windows\\SoftwareDistribution\\Download\\Install\\" 
+  or ParentProcessName !has "\\AppData\\Local\\Temp\\mpam-")
+  | where not(NewProcessName matches regex @"\\Windows\\Microsoft.NET\\(Framework|Framework64)\\v[0-9].[0-9].[0-9]*\\(csc\.exe|cvtres\.exe|mscorsvw\.exe|ngentask\.exe|ngen\.exe)")
+  | where not(NewProcessName matches regex @"\\WindowsAzure\\GuestAgent_[0-9].[0-9].[0-9]*.[0-9]*_[0-9]*-[0-9]*-[0-9]*_[0-9]*\\") 
+  and not(ParentProcessName matches regex @"\\WindowsAzure\\GuestAgent_[0-9].[0-9].[0-9]*.[0-9]*_[0-9]*-[0-9]*-[0-9]*_[0-9]*\\")
+  | where not(NewProcessName matches regex @"\\[0-9A-Za-z]*\\epplauncher.exe")
+  | where not(NewProcessName matches regex @"\\Packages\\Plugins\\Microsoft\.")
+  | extend path_parts = parse_path(NewProcessName)
+  | extend ProcessPath = tostring(path_parts.DirectoryPath)
+  ;
+  let normalizedProcessPath = processEvents
+  | extend NormalizedProcessPath = ProcessPath
+  // normalize guids
+  | project TimeGenerated, Computer, Account, Process, ProcessPath, 
+  NormalizedProcessPath = replace("[0-9A-Fa-f]{8}[-][0-9A-Fa-f]{4}[-][0-9A-Fa-f]{4}[-][0-9A-Fa-f]{4}[-][0-9A-Fa-f]{12}", "<guid>", NormalizedProcessPath)
+  // normalize digits away
+  | project TimeGenerated, Computer, Account, Process, ProcessPath, NormalizedProcessPath = replace(@'\d', 'n', NormalizedProcessPath)
```

Comment:
Probably use * wildcard instead of n. output is easily identifiable and does not collide with existing n letters in the process names

Reply:
Yeah, this came from Tim's other query.  Agree that maybe we use a symbol, such as # or ?.  I don't want to use *.  I will go with # as that is more indicative of a number.